File: java/Deepspace/src/deepspace/PowerEfficientSpaceStation.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author alvaro
 */
public class PowerEfficientSpaceStation extends SpaceStation{
    private final float EFFICIENCYFACTOR = 1.1f;
    
    PowerEfficientSpaceStation(SpaceStation station){
        super(station);
    }
    
    @Override
    public PowerEfficientSpaceStationToUI getUIversion(){
        return new PowerEfficientSpaceStationToUI(this);
    }
    
    @Override
    public float fire(){
        float result = super.fire()*EFFICIENCYFACTOR;
 
        return result;
    }
    
    @Override
    public float protection(){
        float result = super.protection()*EFFICIENCYFACTOR;
        
        return result;
    }
    
    @Override
    public Transformation setLoot(Loot loot){
        Transformation t = super.setLoot(loot);
        
        if ( t == Transformation.SPACECITY )
            return Transformation.NOTRANSFORM;
        else
            return t;
    }
    
    @Override
    public String toString(){
        return "POWER EFFICIENT SPACE STATION: \n" + super.toString();
    }
}

File: ruby/PowerEfficientSpaceStation.rb
#encoding: utf-8

require_relative 'SpaceStation'
require_relative 'PowerEfficientSpaceStationToUI'

module Deepspace
    class PowerEfficientSpaceStation < SpaceStation
        @@EFFICIENCYFACTOR=1.10
        def initialize(station)
            copy(station)
        end

        def fire
            return super*@@EFFICIENCYFACTOR
        end
      
        def protection
            return super*@@EFFICIENCYFACTOR
        end

        def getUIversion
            return PowerEfficientSpaceStationToUI.new(self)
        end

        def setLoot(loot)
            super
            if loot.efficient
              return Transformation::SPACECITY
            else
              return Transformation::NOTRANSFORM
            end
        end

        def to_s
            return "PowerEfficientSpaceStation: " + super
        end
    end
end

